src/runtime/trace.cc

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ * \file trace.cc
+ * \brief Trace packed function.
+*/
+#include <tvm/runtime/registry.h>
+
+namespace tvm {
+namespace runtime {
+
+TVM_REGISTER_GLOBAL("tvm.trace_runtime")
+.set_body([](TVMArgs args, TVMRetValue *ret) {
+    // The name of the user callback should come as first argument.
+    const char *str = args[0].ptr<const char>();
+    size_t size = args.size();
+    CHECK(size >= 2) << "tvm.trace should has at least 2 arguments";
+    if (str) {
+      // Check the global registry for the user defined callback.
+      auto pf = runtime::Registry::Get(str);
+      if (pf) {
+        if (size == 2) {
+          (*pf)(args[1]);
+        } else if (args.size() == 3) {
+          (*pf)(args[1], args[2]);
+        } else if (args.size() == 4) {
+          (*pf)(args[1], args[2], args[3]);
+        } else if (args.size() == 5) {
+          (*pf)(args[1], args[2], args[3], args[4]);
+        } else if (args.size() == 6) {
+          (*pf)(args[1], args[2], args[3], args[4], args[5]);
+        }
```

Comment:
else throw error?

Reply:
@yzhliu thanks for review, moved the check for the arguments count to the line 16, before the call.